test/quantization/pt2e/test_quantize_pt2e.py

```diff
@@ -1233,15 +1234,51 @@ def forward(self, x):
             torch.ops.aten.conv2d.default,
             torch.ops.quantized_decomposed.quantize_per_tensor.default,
         ]
-        example_inputs = (torch.randn(1, 3, 3, 3),)
-        self._test_quantizer(
-            M().eval(),
+        example_inputs = (torch.randn(1, 3, 3, 3, dtype=dtype),)
+        m = self._test_quantizer(
+            M(dtype).eval(),
             example_inputs,
             quantizer,
             node_occurrence,
             node_list,
         )
 
+        # Assumption: topologically sorted graph, single user for each quantize_per_tensor node
+        # Assumption is good enough for the simple model used for testing at present
+        def verify_quant_dequant_iotypes(m, activation_dtype, activation_quant_dtype):
+            quant_node = None
+            for node in m.graph.nodes:
+                if (
```

Comment:
nit: I think it might be more robust and easier to understand to
1. check for dequantize node
2. traverse back to find the corresponding quantize node
    (quant_node = dequant_node.args[0])
4. do all the checks for quant and dequant node